src/components/vm/disks/diskAdd.jsx

```diff
@@ -583,16 +585,35 @@ export class AddDiskModalBody extends React.Component {
             storagePool = storagePools.find(pool => pool.name === this.state.storagePoolName);
             volume = storagePool.volumes.find(vol => vol.name === this.state.existingVolumeName);
             isVolumeUsed = getStorageVolumesUsage(vms, storagePool);
+        } else if (this.state.mode === CUSTOM_PATH) {
+            optionallyResolveFormat = cockpit.spawn(["head", "--bytes", "3", this.state.file], { binary: true, err: "message" })
+                    .then(output => {
+                        let magic = "";
+                        output.map(m => { magic += String.fromCharCode(m) });
+                        switch (magic) {
+                        case "QFI":
```

Comment:
According to QCOW specification [docs](https://forensicswiki.xyz/wiki/index.php?title=QCOW_Image_Format), the magic is first 4 bytes: `QFI\xfb` (or `QFI.` in ASCII)

Reply:
Hmm that's not what I see:

```
[jelle@t14s][~/projects/cockpit-machines]%head --bytes=4 ~/Downloads/disk.qcow2
QFI%
```

```
magic
'QFIû'
output
Uint8Array(4) [81, 70, 73, 251, buffer: ArrayBuffer(4), byteLength: 4, byteOffset: 0, length: 4, Symbol(Symbol.toStringTag): 'Uint8Array']
```

So that's strange? I've grabbed the QFI from https://github.com/cockpit-project/bots/blob/main/machine/machine_core/machine_virtual.py#L321 hmm

Reply:
It's not `.` in ASCII. It's really `\xfb`, so you need to compare it as binary (that's why I said the channel must be binary)